File: codingTest/programmers/정렬/가장 큰 수.js
console.log(solution([12,1213]));

function solution(numbers) {
    var answer = numbers.map(v=>v+'')
    .sort((a,b) => (b+a)*1 - (a+b)*1)
    .join('');

    return answer[0]==='0'?'0':answer;
    // numbers.sort((a,b) => { 
    //     let A = String(a);
    //     let B = String(b);
    //     if (A.length == B.length && A.charAt(0) != B.charAt(0)) 
    //         return B.charAt(0) - A.charAt(0);
    //     else {
    //         let x = 0;
    //         let Aflag = false;
    //         let Bflag = false;
    //         let y = 0;
    //         while (true) {
    //             if (A.charAt(x) != B.charAt(y)) {        
    //                 //console.log(`${B} ${B.charAt(x)} ssv ${A} ${ A.charAt(y)}`);               
    //                 return (B.charAt(y) - A.charAt(x));
    //             }
    //             else {
    //                 if (x < A.length-1)
    //                     x++;
    //                 else {
    //                     x=0;
    //                     Aflag = true;
    //                 }
    //                 if (y < B.length-1) 
    //                     y++;
    //                 else {
    //                     y=0;
    //                     Bflag = true;
    //                 }
    //             }
    //             if (Aflag == true && Bflag == true) {
    //                 return (B.charAt(y) - A.charAt(x));
    //             }
    //         }
    //     }
    // })
    // answer = numbers.join('');
    // if (parseInt(answer)==0)
    //      return '0'
    // else
    //      return answer;
}

File: codingTest/programmers/구현/가장 큰 수.go
// https://programmers.co.kr/learn/courses/30/lessons/42746?language=go
// 가장 큰 수.go

import (
	"sort"
	"strconv"
)

func solution(numbers []int) string {
	answer := ""
	sort.Slice(numbers, func(i, j int) bool {
		a, _ := strconv.Atoi(strconv.Itoa(numbers[i]) + strconv.Itoa(numbers[j]))
		b, _ := strconv.Atoi(strconv.Itoa(numbers[j]) + strconv.Itoa(numbers[i]))
		if a > b {
			return true
		} else {
			return false
		}
	})
	for i, number := range numbers {
		if i == 0 && number == 0 {
			return "0"
		}
		answer += strconv.Itoa(number)
	}
	return answer
}